includes/admin/import/class-batch-import-downloads.php

```diff
@@ -0,0 +1,459 @@
+
+					}
+
+				}
+
+				update_post_meta( $download_id, '_variable_pricing', 1 );
+				update_post_meta( $download_id, 'edd_variable_prices', $variable_prices );
+
+			}
+
+		}
+
+	}
+
+	private function set_files( $download_id = 0, $files = array() ) {
+
+		if( ! empty( $files ) ) {
+
+			$download_files = array();
+			foreach( $files as $file ) {
+
+				$download_files[] = array( 'file' => $file, 'name' => basename( $file ) );
+
+			}
+
+			update_post_meta( $download_id, 'edd_download_files', $download_files );
+
+		}
+
+	}
+
+	private function set_image( $download_id = 0, $image = '' ) {
+
+		$is_url   = false !== filter_var( $image, FILTER_VALIDATE_URL );
+		$is_local = $is_url && false !== strpos( $image, site_url() );
+		$ext      = edd_get_file_extension( $image );
+
+		if( $is_url && $is_local ) {
+
+			// Image given by URL, see if we have an attachment already
+			$attachment_id = attachment_url_to_postid( $image );
+
+		} elseif( $is_url ) {
+
+			if( ! function_exists( 'media_sideload_image' ) ) {
+
+				require_once( ABSPATH . 'wp-admin/includes/file.php' );
+
+			}
+
+			// Image given by external URL
+			$url = media_sideload_image( $image, $download_id, '', 'src' );
+
+			if( ! is_wp_error( $url ) ) {
+
+				$attachment_id = attachment_url_to_postid( $url );
+
+			}
+
+
+		} elseif( false === strpos( $image, '/' ) && edd_get_file_extension( $image ) ) {
+
+			// Image given by name only
+
+			$upload_dir = wp_upload_dir();
+
+			if( file_exists( trailingslashit( $upload_dir['path'] ) . $image ) ) {
+
+				// Look in current upload directory first
+				$file = trailingslashit( $upload_dir['path'] ) . $image;
+
+			} else {
+
+				// Now look through year/month sub folders of upload directory for files with our image's same extension
+				$files = glob( $upload_dir['basedir'] . '/*/*/*{' . $ext . '}', GLOB_BRACE );
+				foreach( $files as $file ) {
+
+					if( basename( $file ) == $image ) {
+
+						// Found our file
+						break;
+
+					}
+
+					// Make sure $file is unset so our empty check below does not return a false positive
+					unset( $file );
+
+				}
+
+			}
+
+			if( ! empty( $file ) ) {
+
+				// We found the file, let's see if it already exists in the media library
+
+				$guid          = str_replace( $upload_dir['basedir'], $upload_dir['baseurl'], $file );
+				$attachment_id = attachment_url_to_postid( $guid );
+
+
+				if( empty( $attachment_id ) ) {
+
+					// Doesn't exist in the media library, let's add it
+
+					$filetype = wp_check_filetype( basename( $file ), null );
+
+					// Prepare an array of post data for the attachment.
+					$attachment = array(
+						'guid'           => $guid,
+						'post_mime_type' => $filetype['type'],
+						'post_title'     => preg_replace( '/\.[^.]+$/', '', $image ),
+						'post_content'   => '',
+						'post_status'    => 'inherit'
+					);
+
+					// Insert the attachment.
+					$attachment_id = wp_insert_attachment( $attachment, $file, $download_id );
```

Comment:
if we support authors, this needs to attach with the author id if its supported by CSV